sound/soc/sof/pcm.c

```diff
@@ -707,7 +708,15 @@ static int sof_pcm_dai_link_fixup(struct snd_soc_pcm_runtime *rtd,
 		/* do nothing for HDA dai_link */
 		break;
 	case SOF_DAI_INTEL_ALH:
-		/* do nothing for ALH dai_link */
+		for_each_dpcm_fe(rtd, 0, dpcm) {
```

Comment:
The '0' used in this macro looks suspect. This means we deal with playback but we also set the trigger for the capture part. Should this be using rtd->substream->stream and only set the relevant triggeristream]?

Reply:
@plbossart @RanderWang I am getting a bit nervous about this loop here:
dpcm_be_dai_hw_params()
   for_each_dpcm_be()
       be_hw_params_fixup()
           for_each_dpcm_fe()

This makes my head spin!

Reply:
@RanderWang why not just set the trigger order in sof_pcm_prepare() for the FE ?

Reply:
I also dont feel comfortable about fixing up the FE dailink's trigger order in a callback meant for BW DAI fixup.

Reply:
@plbossart do you think it makes sense to set this for FE's connected to the ALH BE in intel_hw_params() in drivers/soundwire/intel.com instead of here?

Reply:
@ranj063  thanks for your review. I scanned driver and also tried to find a suitable point to change this sequence.  We changed sequence in sof, so it is easier to understand why we adjust it in sof also. It is difficult to connect driver/soundwire to sound/soc/sof

Reply:
@RanderWang is this truly a SOF FW problem? or a soundwire-specific problem?

Reply:
@ranj063  Not a soundwire-specific problem. the sequence is not aligned with close source driver. you could check close source driver.

Reply:
And this is a fixup for sdw BE so it is suitable here.

Reply:
> @ranj063 Not a soundwire-specific problem. the sequence is not aligned with close source driver. you could check close source driver.

@RanderWang which closed source driver? I thought SOF is the only driver that supports soundwire for Intel platforms?

Reply:
win10 also works on intel platform.